### PR TITLE
[jsonlog] This patch add missing log information that we could find in

### PR DESCRIPTION
**Pull request by darold**
csvlog and with new information available in recent PostgreSQL version.
Here are the new information logged and the corresponding json fields:

  - Line number => "line_number"
  - PS display => "ps_display"
  - session start timestamp => "session_start"
  - internal pos => "internal_pos" (only if an internal query is  logged)
  - backend type => "backend_type"
  - leader PID => "leader_pid"
  - query id => "query_id"

### CODE REVIEW

**Comment by michaelpq** [comment]
Thanks Gilles. I'll look at that.

File: jsonlog/jsonlog.c [hunk 1]
```diff
@@ -193,6 +205,24 @@ setup_formatted_log_time(void)
...
+setup_formatted_start_time(void)
+{
+	pg_time_t       stamp_time = (pg_time_t) MyStartTime;
+
+	/*
+	 * Note: we expect that guc.c will ensure that log_timezone is set up (at
+	 * least with a minimal GMT value) before Log_line_prefix can become
+	 * nonempty or CSV mode can be selected.
+	 */
+	pg_strftime(formatted_start_time, FORMATTED_TS_LEN,
```
Comment (michaelpq) [inline jsonlog/jsonlog.c:221]: Shouldn't this generate a timestamp in the same format as setup_formatted_log_time() to have something machine-readable? Note commit f7960f1.

File: jsonlog/jsonlog.c [hunk 2]
```diff
@@ -150,7 +162,7 @@ write_console(char *data, int len)
 	int		 rc;
 
 	Assert(len > 0);
-	rc = write(fd, data, PIPE_HEADER_SIZE + len);
+	rc = write(fd, data, len);
```
Comment (michaelpq) [inline jsonlog/jsonlog.c:165]: What's this for?

**Comment by darold** [comment]
To fix #27, the addition of PIPE_HEADER_SIZE adds garbage to the log sent to console at startup. You can try by using log_destination to stderr, logging_collector to on, then starting PG like `/usr/local/pgsql/bin/pg_ctl -D /usr/local/pgsql/data -l logfile start`. File logfile will write one line with garbage at end. My understanding is that PIPE_HEADER_SIZE is to be used in write_pipe_chunks() not to write to console.

And actually write_console() in PG code do not include this extra PIPE_HEADER_SIZE so I guess I am right.

File: jsonlog/jsonlog.c (hunk 1 above)
Comment (darold) [reply jsonlog/jsonlog.c:221]: Honestly I have just copy/paste the function from elog.c, I guess that we want the same output.

**Comment by michaelpq** [comment]
> And actually write_console() in PG code do not include this extra PIPE_HEADER_SIZE so I guess I am right.

It may be better to look at that separately then.

**Comment by darold** [comment]
Removed

**Comment by michaelpq** [comment]
Okay, I have applied 18f1a6a and f6e6686, to respectively add the missing fields and fix the issue with the junk data sent to stderr because of the incorrect length that was added. I still need to look at the compatibility. Why do you need compatibility down to 9.2? I tend to minimize the maintenance of this module, and that means testing across many versions, some of them not supported by the community for years now.

**Comment by darold** [comment]
Because the code by itself supports 9.2 and I'm pretty sure that in real life there is 9.2 who still be there for several years :-)

**Comment by michaelpq** [comment]
> Because the code by itself supports 9.2 and I'm pretty sure that in real life there is 9.2 who still be there for several years :-)

I may just cut that down to 9.6 at the end, to stick with upstream, and I don't have older branches compiled at hand.

**Comment by darold** [comment]
Dommage, it's not a big effort, but on my side I use latest so I'm not impacted.

**Comment by michaelpq** [comment]
Okay. Done the compatibility bits with 6a5d795.

**Pull request closed without merging**